src/snapred/backend/service/ReductionService.py

```diff
@@ -85,42 +95,67 @@ def validateReduction(self, request: ReductionRequest):
+                continueFlags |= ContinueWarning.Type.MISSING_NORMALIZATION
+                warningMessages.append("Normalization is missing, continuing with artificial normalization step.")
+            # check if a diffraction calibration is present
+            if not self.dataFactoryService.calibrationExists(request.runNumber, request.useLiteMode):
+                continueFlags |= ContinueWarning.Type.MISSING_DIFFRACTION_CALIBRATION
+                warningMessages.append("Diffraction calibration is missing, continuing with uncalibrated mode.")
+
+            # remove any continue flags that are present in the request by xor-ing with the flags
+            if request.continueFlags:
+                continueFlags = continueFlags ^ (request.continueFlags & continueFlags)
+
+            if continueFlags:
+                detailedMessage = "\n".join(warningMessages)
+                raise ContinueWarning(
+                    f"The reduction cannot proceed due to missing data:\n{detailedMessage}\n", continueFlags
+                )
 
-        # ... ensure separate continue warnings ...
-        continueFlags = ContinueWarning.Type.UNSET
+            # ... ensure separate continue warnings ...
+            continueFlags = ContinueWarning.Type.UNSET
 
-        # check that the user has write permissions to the save directory
-        if not self.checkWritePermissions(request.runNumber):
-            continueFlags |= ContinueWarning.Type.NO_WRITE_PERMISSIONS
+            # check that the user has write permissions to the save directory
+            if not self.checkWritePermissions(request.runNumber):
+                continueFlags |= ContinueWarning.Type.NO_WRITE_PERMISSIONS
 
-        # remove any continue flags that are present in the request by xor-ing with the flags
-        if request.continueFlags:
-            continueFlags = continueFlags ^ (request.continueFlags & continueFlags)
+            # remove any continue flags that are present in the request by xor-ing with the flags
+            if request.continueFlags:
+                continueFlags = continueFlags ^ (request.continueFlags & continueFlags)
 
-        if continueFlags:
-            raise ContinueWarning(
-                f"<p>It looks like you don't have permissions to write to "
-                f"<br><b>{self.getSavePath(request.runNumber)}</b>,<br>"
-                + "but you can still save using the workbench tools.</p>"
-                + "<p>Would you like to continue anyway?</p>",
-                continueFlags,
-            )
+            if continueFlags:
+                raise ContinueWarning(
+                    f"<p>It looks like you don't have permissions to write to "
+                    f"<br><b>{self.getSavePath(request.runNumber)}</b>,<br>"
+                    + "but you can still save using the workbench tools.</p>"
+                    + "<p>Would you like to continue anyway?</p>",
+                    continueFlags,
+                )
```

Comment:
[This](https://github.com/neutrons/SNAPRed/pull/463/files#diff-8f698031b2852fad83d059775e7f13b373b64302000cb3f5dc92d1556c0f6ed0R99-R116) and [this](https://github.com/neutrons/SNAPRed/pull/463/files#diff-8f698031b2852fad83d059775e7f13b373b64302000cb3f5dc92d1556c0f6ed0R99-R116) are duplicate code.  Please break this method out into smaller sub methods to reduce its complexity and the amount of nesting.
i.e. [this](https://github.com/neutrons/SNAPRed/pull/463/files#diff-8f698031b2852fad83d059775e7f13b373b64302000cb3f5dc92d1556c0f6ed0R99-R116) should be its own method, and [this](https://github.com/neutrons/SNAPRed/pull/463/files#diff-8f698031b2852fad83d059775e7f13b373b64302000cb3f5dc92d1556c0f6ed0R118-R138) should be another method, and those two should be substituted into the above method like so(I reorged it so we didnt need to check write perms twice):
```
if request.artificialNormalization is None:
    self._validateCalibrationAndNormalizationExists(request)

self._validateWritePermissions(request)

```